app/models/report.rb

```diff
@@ -478,8 +613,217 @@ def self.report_trending_search(report)
+      mod_data[row.user_id][:flag_count] = row.flag_count if row.user_id
+    end
+
+    DB.query(topic_count_query).each do |row|
+      mod_data[row.user_id][:topic_count] = row.topic_count if row.user_id
+    end
+
+    DB.query(post_count_query).each do |row|
+      mod_data[row.user_id][:post_count] = row.post_count if row.user_id
+    end
+
+    report.data = mod_data.values
+  end
+
+  def self.report_flags_status(report)
+    report.modes = [:table]
+
+    report.labels = [
+      { properties: [:action_type], title: I18n.t("reports.flags_status.labels.flag") },
+      { type: :link, properties: [:staff_username, :staff_url], title: I18n.t("reports.flags_status.labels.assigned") },
+      { type: :link, properties: [:poster_username, :poster_url], title: I18n.t("reports.flags_status.labels.poster") },
+      { type: :link, properties: [:flagger_username, :flagger_url], title: I18n.t("reports.flags_status.labels.flagger") },
+      { type: :seconds, properties: [:response_time], title: I18n.t("reports.flags_status.labels.time_to_resolution") }
+    ]
+
+    report.data = []
+
+    flag_types = PostActionType.flag_types_without_custom
+
+    sql = <<~SQL
+    SELECT pa.post_action_type_id,
+    p.user_id AS poster_id,
+    pa.post_action_type_id,
+    pa.created_at,
+    pa.agreed_at,
+    pa.disagreed_at,
+    pa.deferred_at,
+    pa.agreed_by_id,
+    pa.disagreed_by_id,
+    pa.deferred_by_id,
+    pa.user_id AS flagger_id,
+    (select u.username FROM users u WHERE u.id = pa.user_id) AS flagger_username,
+    COALESCE(pa.disagreed_at, pa.agreed_at, pa.deferred_at, NULL) AS responded_at,
+    COALESCE(pa.agreed_by_id, pa.disagreed_by_id, pa.deferred_by_id, NULL) AS staff_id,
+    (SELECT u.username FROM users u WHERE u.id = COALESCE(pa.agreed_by_id, pa.disagreed_by_id, pa.deferred_by_id, null)) AS staff_username,
+    (SELECT u.username FROM users u WHERE u.id = p.user_id) as poster_username
+    FROM post_actions pa
+    JOIN posts p
+    ON p.id = pa.post_id
+    WHERE pa.post_action_type_id = ANY(ARRAY#{PostActionType.flag_types_without_custom.values})
+    AND pa.created_at >= '#{report.start_date}'
+    AND pa.created_at <= '#{report.end_date}'
+    ORDER BY pa.created_at DESC
+    LIMIT 20
+    SQL
+
+    DB.query(sql).each do |row|
+      data = {}
+      data[:action_type] = flag_types.key(row.post_action_type_id).to_s
+      data[:staff_username] = row.staff_username
+      data[:staff_id] = row.staff_id
+      if row.staff_username && row.staff_id
+        data[:staff_url] = "/admin/users/#{row.staff_id}/#{row.staff_username}"
+      end
+      data[:poster_username] = row.poster_username
+      data[:poster_id] = row.poster_id
+      data[:poster_url] = "/admin/users/#{row.poster_id}/#{row.poster_username}"
+      data[:flagger_id] = row.flagger_id
+      data[:flagger_username] = row.flagger_username
+      data[:flagger_url] = "/admin/users/#{row.flagger_id}/#{row.flagger_username}"
```

Comment:
I think all these URLs should be generated on the client side.

Reply:
I can ’t actually do this, as I want reports to be "standalone" and it would force us to carry more logic. Maybe I can reconsider this in the future.